src/db/write.rs

```diff
@@ -193,8 +193,18 @@ impl<'a> Write<'a> {
         key_prefix: &[u8],
         json_pointer: &str,
     ) -> Result<(), CreateIndexError> {
-        // TODO: This should be a nop if the specified index already exists.
         use CreateIndexError::*;
+
+        // Check to see if the index already exists.
+        if let Some(index) = self.indexes.get(name) {
+            if index.meta.definition.key_prefix == key_prefix
```

Comment:
Use `PartialEq` instead so that this generalizes as definition becomes more sophisticated.

Reply:
I'm not sure what that would mean in this context. (And I'm also not sure why an index definition would not be reflexive (Eq)).

Reply:
You're right, I was reading this too fast.

Reply:
Take that back, I was right. This is what I meant:

https://github.com/rocicorp/repc/pull/212/commits/773b2319be24d28124850441e85b79831604fdfc

Reply:
(This way if we add fields to definition in the future we won't have to remember to add them to this path).